src/Bicep.Core/Emit/ExpressionEmitter.cs

```diff
@@ -91,6 +91,16 @@ public void EmitExpression(SyntaxBase syntax)
             }
         }
 
+        public void EmitResourceName(SyntaxBase resourceNameSyntax, SyntaxBase? indexExpression, SyntaxBase newContext)
```

Comment:
[nit] rename to something more general (`EmitExpression`) as it doesn't just apply to resource names? Same with the param `resourceNameSyntax` and `resourceName` variable.

Reply:
renamed